### PR TITLE
Simplify Welcome Screen and use card-based layout for examples

### LINKED ISSUE #5452: Card based example gallery in the application welcome screen
New design here: https://www.figma.com/file/U85f12k8Tw2ArD4N61cdtQ/Marketing-material?node-id=2272%3A42422

### PR DESCRIPTION
**Pull request by abey79**
### What

* Closes #5452 

Note: the "Source code" links are functional but disabled, as the example manifest must be updated to include the corresponding URL, see follow-up issues.

Follow-up:
- #5700 
- #5701 

Before/After:

image

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5699/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5699/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5699/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5699)
- [Docs preview](https://rerun.io/preview/6766af1787603295c4d4970a1c98849677409942/docs) 
- [Examples preview](https://rerun.io/preview/6766af1787603295c4d4970a1c98849677409942/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by emilk** [comment]
The hover effect is way to subtle imho (at least in a bright room). It would be good if the full rectangle got highlighted somehow e.g. by:
* Making it bigger
* Making it brighter
* Adding an outline

subtle-hover

File: crates/re_ui/src/design_tokens.rs [hunk 1]
```diff
@@ -84,21 +84,21 @@ fn apply_design_tokens(ctx: &egui::Context) -> DesignTokens {
...
+        egui::FontId::proportional(13.0),
     );
     egui_style.text_styles.insert(
         ReUi::welcome_screen_body(),
-        egui::FontId::proportional(13.0),
+        egui::FontId::proportional(15.0),
     );
+    egui_style
+        .text_styles
+        .insert(ReUi::welcome_screen_tag(), egui::FontId::proportional(9.5));
```
Comment (emilk) [inline crates/re_ui/src/design_tokens.rs:101]: I think this might be too small.

**Review by emilk — APPROVED** [review]
Looks great, but I think we need a better hover-effect. Adding an `Color32::from_additive_luminance(32)` or so rect fill on it should do the trick

File: crates/re_ui/data/icons/github.png
[the remark this answers is not in the document]
Comment (abey79) [reply crates/re_ui/data/icons/github.png:1]: Pixel alignment wasn't great, actually. Fixed and updated.

File: crates/re_viewer/src/ui/welcome_screen/example_section.rs [hunk 2]
```diff
@@ -368,129 +436,91 @@ impl ExampleSection {
...
 }
 
 fn example_thumbnail(
     ui: &mut Ui,
     example: &ExampleDesc,
-    thumbnail_size: egui::Vec2,
-    hovered: bool,
+    column_width: f32,
+    width: f32,
+    height: f32,
```
[the remark this answers is not in the document]
Comment (abey79) [reply crates/re_viewer/src/ui/welcome_screen/example_section.rs:447]: Actually, I cleaned that up further. Those dimension were already part of that function's first argument.

File: crates/re_viewer/src/ui/welcome_screen/example_section.rs [hunk 3]
```diff
@@ -499,7 +529,36 @@ fn example_tags(ui: &mut Ui, example: &ExampleDesc) {
...
+            .add_enabled(
+                source_url.is_some(),
+                egui::Button::image_and_text(re_ui::icons::GITHUB.as_image(), "Source code"),
+            )
+            .on_disabled_hover_text("Source code is not available for this example")
+            .clicked()
+        {
+            if let Some(source_url) = source_url {
+                ui.ctx().output_mut(|o| {
+                    o.open_url = Some(egui::output::OpenUrl {
```
[the remark this answers is not in the document]
Comment (abey79) [reply crates/re_viewer/src/ui/welcome_screen/example_section.rs:556]: Neat! Fixed it in some other places.

File: crates/re_ui/src/design_tokens.rs (hunk 1 above)
Comment (abey79) [reply crates/re_ui/src/design_tokens.rs:101]: I strictly followed the source design 🤷🏻  Bumped to 10.5.

**Comment by abey79** [comment]
> Looks great, but I think we need a better hover-effect. Adding an `Color32::from_additive_luminance(32)` or so rect fill on it should do the trick

I did that, albeit with a value of 25, it's pretty nice 👍🏻

**Comment by emilk** [comment]
Nice, the new hover effect works great!

**Pull request merged**